functions/HttpTriggerAPIUsersId/__init__.py

```diff
@@ -5,189 +5,259 @@
+        assert 'email' in user_req_body, 'User request body did not contain field: "email"'
     except AssertionError as user_req_body_content_error:
         logging.error(
-            "User request body did not contain the necessary fields!")
-        return func.HttpResponse(user_req_body_content_error.args[0], status_code=400)
-    logging.debug("User request body contains all the necessary fields!")
-
-    # Unpack user data
-    firstName = user_req_body["firstName"]
-    lastName = user_req_body["lastName"]
-    email = user_req_body["email"]
-
-    # Update user in DB
-    update_user_query = "UPDATE dbo.users SET firstName = ?, lastName = ?, email = ? WHERE userId= ?"
-    logging.debug("Executing query: " + update_user_query)
-    cursor.execute(update_user_query,
-                   (firstName, lastName, email, user_id))
-    logging.debug("User was updated successfully!.")
-    return func.HttpResponse(
-        "User updated",
-        status_code=200
-    )
+            'User request body did not contain the necessary fields!'
+        )
+        return func.HttpResponse(
+          user_req_body_content_error.args[0], status_code=400
+        )
 
+    logging.debug('User request body contains all the necessary fields!')
 
-def delete(cursor, user_id):
-    logging.debug("Attempting to retrieve user by ID and delete the user...")
-    delete_user_query = "DELETE FROM dbo.users  WHERE userId= ?"
-    logging.debug("Executing query: " + delete_user_query)
-    cursor.execute(delete_user_query, (user_id,))
-    logging.debug("User was deleted successfully!.")
-    return func.HttpResponse(
-        "User deleted",
-        status_code=200
-    )
+    with conn.cursor() as cursor:
+        # Unpack user data
+        firstName = user_req_body['firstName']
+        lastName = user_req_body['lastName']
+        email = user_req_body['email']
 
+        # Update user in DB
+        update_user_query = '''
+          UPDATE dbo.users SET firstName = ?,
+          lastName = ?, email = ? WHERE userId= ?
+        '''
 
-def get_user_row(cursor, user_id):
-    cursor.execute(
-        "SELECT * FROM dbo.users WHERE userId= ?", (user_id,))
+        logging.debug('Executing query: ' + update_user_query)
 
-    return cursor.fetchone()
+        cursor.execute(
+          update_user_query, (firstName, lastName, email, user_id)
+        )
 
-#This method caches user_id
-#param: r- redis cache
-#user_id: User IDs that need to cached
-def cache_user_id(r, user_id):
-    key = "users:" + userId
-    try:
-        r.set(key, json.dumps(users), ex= 1200) 
-        logging.info("Caching complete!")
-    except TypeError as e:
-        logging.info("Caching failed")
-        logging.info(e.args[0])
+        logging.debug('User was updated successfully!.')
+
+        return func.HttpResponse('User updated', status_code=200)
+
+
+def patch_user(user_req_body, conn, user_id):
```

Comment:
Patch method at the moment does not update the user correctly. It updates it but not with the correct data